### PR TITLE
add html parser hidden import for botocore

### LINKED ISSUE #1844: test_boto3 and test_botocore fails due to package update (bundled six)
Both Appveyor and Travis are showing failures in test_boto3 and test_botocore due to an update to both packages.
- The exe complains about missing module `html.parser`:

```
  File "/lib/python3.4/site-packages/botocore/vendored/six.py", line 79, in _import_module
    __import__(name)
ImportError: No module named 'html.parser'
LOADER: Back to parent (RC: 255)
LOADER: Doing cleanup
LOADER: Freeing archive status for /test_botocore_onedir_0/dist/test_botocore/test_botocore
```
- module `html.parser` is wrapped in `six` module.
- I think this is somehow connected to the bundled version of `six` module as `botocore.vendored.six`.
- I guess PyInstaller is not able to execute the `six` pre_safe_import hook for module `botocore.vendored.six`. Probly some doing that could fix those tests.

### PR DESCRIPTION
**Pull request by jstewmon**
The dependency on html parser was added to botcore in:
https://github.com/boto/botocore/commit/ec85db75c26d3bb81e1831e81191b98f37de9f27

Fixes #1844

### CODE REVIEW

**Comment by jstewmon** [comment]
@htgoebel The tests are passing for me locally, but on CI, it seems like PyInstaller is not discovering all imports. I've been looking into this for a couple of hours without being able to figure it out, so if you could take a look, I'd appreciate it.

The test script has `import botocore` and `from botocore.session import Session`. `botocore.session` [imports `botocore.credentials`](https://github.com/boto/botocore/blob/1.4.31/botocore/session.py#L26) and `botocore.credentials` [imports `dateutil.parser`](https://github.com/boto/botocore/blob/1.4.31/botocore/credentials.py#L22), but the test is failing on CI with an import error for `dateutil.parser`:

```
Traceback (most recent call last):
  File "/tmp/pytest-of-travis/pytest-0/popen-gw1/test_botocore_onedir_0/test_botocore.py", line 4, in <module>
    from botocore.session import Session
  File "/home/travis/build/pyinstaller/pyinstaller/PyInstaller/loader/pyimod03_importers.py", line 389, in load_module
    exec(bytecode, module.__dict__)
  File "botocore/session.py", line 26, in <module>
  File "/home/travis/build/pyinstaller/pyinstaller/PyInstaller/loader/pyimod03_importers.py", line 389, in load_module
    exec(bytecode, module.__dict__)
  File "botocore/credentials.py", line 22, in <module>
ImportError: No module named dateutil.parser
```

File: PyInstaller/hooks/hook-botocore.py [hunk 1]
```diff
@@ -19,5 +19,23 @@
...
+
+hiddenimports = []
+
+if is_module_satisfies('botocore >= 1.3.30'):
+    if is_py3:
+        hiddenimports.extend(
+            [
+                'html.parser'
+            ]
+        )
```
Comment (htgoebel) [inline PyInstaller/hooks/hook-botocore.py]: This should be either `hiddenimports.append(...)` or just `hiddenimports = [...]`. And please write this on a single line.

File: tests/requirements-libraries.txt [hunk 2]
```diff
@@ -25,7 +25,7 @@ zope.interface  # Required for test_namespace_package
 # numpy 1.11 is broken and does not compile on AppVeyor (Windows)
 numpy==1.10.4
 lxml
-keyring
+keyring==9.0
```
Comment (htgoebel) [inline tests/requirements-libraries.txt]: current version fails with what?

**Comment by htgoebel** [comment]
Thanks for this pull-request, which I'd like to merge. I added some review.

**Comment by htgoebel** [comment]
There is an updated hook for botocore now, see #2304, implementing what requested. Please update your pull-request. This can best be done by pulling current master and rebase you branch on top of it.

**Comment by jstewmon** [comment]
@htgoebel it looks like #2304 duplicates the purpose of this PR by addressing #1844. Is there something else in this PR you want to merge? If so, I would prefer to close this PR and open a new PR for that purpose.

**Comment by htgoebel** [comment]
Yes, your changes in c650c7fd3 and 761856df06 seam to be worth kept. I'd appreciate if you'd move these to a separate PR :-)

**Comment by jstewmon** [comment]
Ok, those changes are on #2352

**Pull request closed without merging**